### ISSUE TITLE
ValueError on RFB buffer when "Too many security failures" received

### ISSUE THREAD

**Issue by freetomik** [sibson/vncdotool#110]
According to [1] VNC server blocks IP address after five unsuccessful connection attempts from client. In this situation I got exception from vncdotool:
```
File "/usr/lib/python2.7/site-packages/vncdotool/rfb.py", line 133, in _handleInitial
        version_server = float(buffer[3:-1].replace(b'0', b''))
        exceptions.ValueError: invalid literal for float(): 3.3
```
After debugging I found out, that `buffer` contains `'RFB 003.003\n\x00\x00\x00\x00\x00\x00\x00\x1aToo many security failures'` instead of `'RFB 003.003\n'`.

Possible fix I tried could be
```
version_server = float(buffer.split()[1].replace(b'0', b''))
```

Tried with tigervnc.
[1] https://support.realvnc.com/knowledgebase/article/View/244/1/too-many-security-failures

**Comment by hilbix** [comment]
For me this even looks like a couple of bugs in following code.  (A sloppy quick fix would be to replace the `-1` by `12` on L133.  However this leaves open some further bugs.  See below.)

https://github.com/sibson/vncdotool/blob/e133a8916efaa0f5ed421e0aa737196624635b0c/vncdotool/rfb.py#L130-L142

1. [The RFB-header is defined to be exactly 12 bytes][1].  Hence, this routine should wait for at least 12 bytes to arrive and then, immediately, check if it is a proper `RFB ###.###\n`, and raise some error if it is not.  
Please note that the RFB protocol does not start with just `RFB`.  It starts with the hex sequence 52 46 42 20 which is `RFB` followed by an SPC.  And the 8th byte must be a dot (hex 2e) and the last 12th byte must be an LF (hex 0a).  See [RFC 6143][1].  If you receive something completely different, the code should catch that and refuse.

2. The code loops until the `in` finds some `\n`.  But if you happen to hit something like `socat tcp-listen:5901,reuseaddr,fork file:/dev/null`, which never outputs an LF, this will quickly exhaust all of your precious CPU, RAM and bandwidth, alltogether with a breeze!

3. If you look at L142, the code expects, that more than 12 bytes can arrive in the first packet.  However the code leading there apparently fails to properly handle this case if it is an RFB server.  Also this line skips 12 bytes, even that the above lines would handle shorter or longer headers (which violate the protocol), too.

4. In contrast, if it is not an RFB server (the first 3 bytes are not `RFB`) then (silently!) RFB protocol 3.3 is assumed.  WTF?  RFB is even detected when the first packet  just contains a single `\n`!

[1]: https://tools.ietf.org/html/rfc6143#section-7.1.1

**Comment by pmhahn** [comment]
bc93eb9739477aa8f6d3075b3c1d5675a137d1b8 changed to server version parsing, which is part of #229. The issue is probably not fixed completely, but should have improved at least:
- `float`s are no longer used but `int`s
- a _regular expression_ is now used for matching

```
RFB 003.003\n → valid HEADER
\x00\x00\x00\x00 → AuthTypes.INVALID
\x00\x00\x00\x1a → length-of-message = 26 bytes
Too many security failures  → message
```

What's missing is the `else` case when the regular expression does *not* match: In that case the connection should probably be terminated. The current code using the _regular expression_ has the following issues:
- it never calls `transport.loseConnection()`
- the code should also loose the connection if the already received answer will never match.

This should be fixed in my MR #250 pending merge.

**Comment by sibson** [comment]
Has this been resolved with recent changes?
